pkg/proc/bininfo.go

```diff
@@ -974,19 +1007,15 @@ func (bi *BinaryInfo) setGStructOffsetElf(image *Image, exe *elf.File, wg *sync.
 			break
 		}
 	}
-	if tlsg == nil {
-		bi.gStructOffset = ^uint64(8) + 1 // -8
-		return
-	}
 	var tls *elf.Prog
 	for _, prog := range exe.Progs {
 		if prog.Type == elf.PT_TLS {
 			tls = prog
 			break
 		}
 	}
-	if tls == nil {
-		bi.gStructOffset = ^uint64(8) + 1 // -8
+	if tlsg == nil || tls == nil {
+		bi.gStructOffset = ^uint64(bi.Arch.PtrSize()) + 1 //-ptrSize + 1
```

Comment:
The comment is wrong, it's `// -ptrSize` not `-ptrSize + 1` (it's taking the 2s complement of 8).

Reply:
Oh, sorry for this mistake

Reply:
Done